### PR TITLE
replace all API with mutiny reactive

### PR DESCRIPTION
**Pull request by shokakucarrier**

### CODE REVIEW

**Review by ruhan1 — COMMENTED** [review]
This looks good! Moving the logic from Resource to Service class is a good practice. 
I would suggest one change to the create(). I don't know if the controller.renderArchive is async (off the work to backend thread). If not, I'd suggest doing so. You may use the Quarkus EventBus to pass the work to backend thread.

**Comment by shokakucarrier** [comment]
> 
> 
> This looks good! Moving the logic from Resource to Service class is a good practice.
> I would suggest one change to the create(). I don't know if the controller.renderArchive is async (off the work to backend thread). If not, I'd suggest doing so. You may use the Quarkus EventBus to pass the work to backend thread.

Nope, it is not, I'll take a look on that later, thx

**Review by yma955 — CHANGES_REQUESTED** [review]
I don't think it's necessary to do this refactor! The original resource logic is simple enough,  all are controlled in the controller. No need to extract, at least, by now. If you want to do something about Uni, just doing that is enough.

**Comment by shokakucarrier** [comment]
> I don't think it's necessary to do this refactor! The original resource logic is simple enough, all are controlled in the controller. No need to extract, at least, by now. If you want to do something about Uni, just doing that is enough.

Wha.... but why don’t we follow the design patterns quarkus officially follows. Why do we put logics into API announcing codes anyway.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 1]
```diff
@@ -109,88 +105,100 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
             Optional<File> archive = controller.generateArchive( content );
             if ( archive.isEmpty() )
             {
-                final String message = "Failed to get downloaded contents for archive.";
-                logger.error( message );
-                return responseHelper.fromResponse( message );
+                return sendError( 500, "Failed to get downloaded contents for archive." );
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: Not recommend to use magical number. Maybe Status.INTERNAL_SERVER_ERROR.getStatusCode() here and following?

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 2]
```diff
@@ -109,88 +105,100 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
         try
         {
             controller.deleteArchive( buildConfigId );
         }
         catch ( final IOException e )
         {
-            final String message = "Failed to delete historical archive for build config Id " + buildConfigId;
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
+            return Uni.createFrom()
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: sendError(NOT_FOUND.getStatusCode(), "") here?

**Review by ligangty — COMMENTED** [review]
Generally LGTM except some magical numbers.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java (hunk 1 above)
Comment (shokakucarrier) [reply src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: I've switched to sendResponse(Status status, String message), we don't have to use int to represent status code anymore

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 3]
```diff
@@ -71,35 +71,29 @@
     @Inject
     ArchiveController controller;
 
-    @Inject
-    ResponseHelper responseHelper;
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:75]: @shokakucarrier You need to keep this, wrap the functions about response/header in it.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 4]
```diff
@@ -71,35 +71,29 @@
...
-                return responseHelper.fromResponse( message );
+                return sendResponse( Status.INTERNAL_SERVER_ERROR, "Failed to read historical content which is empty." );
             }
         }
         catch ( final IOException e )
         {
-            final String message = "Failed to read historical content file from request body.";
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
+            return sendResponse( Status.INTERNAL_SERVER_ERROR, "Failed to read historical content file from request body." );
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:96]: @shokakucarrier sendResponse doesn't track exception log output.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 5]
```diff
@@ -109,88 +103,95 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
     @GET
-    public Response get( final @PathParam( "buildConfigId" ) String buildConfigId, final @Context UriInfo uriInfo )
+    public Uni<Response> get( final @PathParam( "buildConfigId" ) String buildConfigId, final @Context UriInfo uriInfo )
     {
-        Response response = null;
+        InputStream inputStream = null;
+
         try
         {
-            Optional<File> target = controller.getArchiveInputStream( buildConfigId );
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:150]: @shokakucarrier This rename is not necessary

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 6]
```diff
@@ -109,88 +103,95 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
-            if ( target.isPresent() )
+            Optional<File> file = controller.getArchiveInputStream( buildConfigId );
+            if ( file.isPresent() )
             {
-                InputStream inputStream = FileUtils.openInputStream( target.get() );
-                final ResponseBuilder builder = Response.ok( new TransferStreamingOutput( inputStream ) );
-
-                response = buildWithHeader( builder, buildConfigId );
+                inputStream = FileUtils.openInputStream( file.get() );
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:144]: @shokakucarrier This is really redundant, wrapping the output in builder then returning in Response, please change back.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 7]
```diff
@@ -109,88 +103,95 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
         try
         {
             controller.deleteArchive( buildConfigId );
         }
         catch ( final IOException e )
         {
-            final String message = "Failed to delete historical archive for build config Id " + buildConfigId;
-            logger.error( message, e );
-            return responseHelper.fromResponse( message );
+            return sendResponse( Status.NOT_FOUND, "");
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:168]: @shokakucarrier This is not 404 error, there might be many reasons causing error on deletion.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 8]
```diff
@@ -109,88 +103,95 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
         }
-        return noContent().build();
+
+        return sendResponse( Status.NO_CONTENT, "" );
     }
 
-    private Response buildWithHeader( ResponseBuilder builder, final String buildConfigId )
+    private Uni<Response> buildWithHeader( Response.ResponseBuilder builder, final String buildConfigId )
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:174]: @shokakucarrier it's not preferred to use Response.ResponseBuilder

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 9]
```diff
@@ -109,88 +103,95 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
+    {
+        logger.error( message );
+        return Uni.createFrom().item( message )
+                             .onItem()
+                             .transform( re -> Response.status( status ).type( TEXT_PLAIN ).entity( re ) )
+                             .onItem()
+                             .transform( Response.ResponseBuilder::build );
+    }
+
 }
```
Comment (yma955) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:197]: @shokakucarrier Converting the ResponseHelper into a method is not suggested, we might extend further in the helper class, you could read Indy, Indy-miro* codebases, it's widely used.
Actually, this is not needed, you could just add Uni in previous helper related methods.

1. Please wrap the method in helper class, then call through inject
2. Don't expose INTERNAL_SERVER_ERROR in resource class, helper is doing this.

**Review by yma955 — CHANGES_REQUESTED** [review]
@shokakucarrier I still don't see any change based on the last comments, maybe that's because I set the reviews to pending, please review the comments I send them out.

**Comment by shokakucarrier** [comment]
I'm not gonna make this patch anymore, do what you would like to, and good luck

**Pull request closed without merging**